### PR TITLE
add get cluster's namespace interface; use nginx to connect frontend

### PR DESCRIPTION
**Pull request by chenkaiyue**

### CODE REVIEW

File: deploy/craned/deployment.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ spec:
           operator: Exists
       containers:
         - name: craned
-          image: docker.io/gocrane/craned:v0.2.0
+          image: ccr.ccs.tencentyun.com/chenky/crane:022101
```
Comment (kitianFresh) [inline deploy/craned/deployment.yaml:38]: Do not modify the image.

File: deploy/craned/deployment.yaml [hunk 2]
```diff
@@ -49,6 +49,17 @@ spec:
               readOnly: true
             - name: config
               mountPath: /tmp/recommendation-config
+          readinessProbe:
+            httpGet:
+              path: /api/healthz
+              port: 8082
+              scheme: HTTP
+        - image: gcr.io/kubecost1/frontend:prod-1.90.0-rc.0
```
Comment (kitianFresh) [inline deploy/craned/deployment.yaml:57]: Please replace this image

File: deploy/craned/service.yaml [hunk 3]
```diff
@@ -11,3 +11,31 @@ spec:
...
+  namespace: crane-system
+spec:
+  ports:
+    - port: 8082
+      protocol: TCP
+      targetPort: 8082
+  selector:
+    app: craned
+---
+apiVersion: v1
```
Comment (kitianFresh) [inline deploy/craned/service.yaml:28]: I think we should only expose one service name crane. the service will expose multiple ports, include crane(which is the crane nginx web endpoint), craned-http (which is craned controller), craned web-http (which is craned web server), craned metrics-http (which is craned metrics)  @qmhu

File: deploy/craned/deployment.yaml (hunk 1 above)
Comment (chenkaiyue) [reply deploy/craned/deployment.yaml]: Ok

Comment (chenkaiyue) [reply deploy/craned/deployment.yaml]: Ok

File: deploy/craned/deployment.yaml (hunk 2 above)
Comment (chenkaiyue) [reply deploy/craned/deployment.yaml]: Just for test

Comment (chenkaiyue) [reply deploy/craned/deployment.yaml]: Just for test

File: deploy/craned/service.yaml (hunk 3 above)
Comment (chenkaiyue) [reply deploy/craned/service.yaml]: Ok

File: deploy/craned/deployment.yaml [hunk 4]
```diff
@@ -49,6 +49,20 @@ spec:
...
+              port: 8082
+              scheme: HTTP
+        - image: docker.io/gocrane/dashboard:v0.2.0
+          imagePullPolicy: IfNotPresent
+          name: dashboard
+          volumeMounts:
+            - mountPath: /etc/nginx/conf.d/
+              name: nginx-conf
+          env:
+          - name: CLUSTER_ID
```
Comment (kitianFresh) [inline deploy/craned/deployment.yaml:64]: What is this used for? the front end use ?

**Pull request merged**